src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/GCStaticsPreInitDataNode.cs

```diff
@@ -0,0 +1,112 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Linq;
+
+using Internal.Text;
+using Internal.TypeSystem;
+
+using Debug = System.Diagnostics.Debug;
+
+namespace ILCompiler.DependencyAnalysis
+{
+    /// <summary>
+    /// Contains all GC static fields for a particular EEType.
+    /// Fields that have preinitialized data are pointer reloc pointing to frozen objects.
+    /// Other fields are initialized with 0.
+    /// We simply memcpy these over the GC static EEType object.
+    /// </summary>
+    public class GCStaticsPreInitDataNode : ObjectNode, IExportableSymbolNode
+    {
+        private MetadataType _type;
+        private List<PreInitFieldInfo> _sortedPreInitFields;
+
+        public GCStaticsPreInitDataNode(MetadataType type, List<PreInitFieldInfo> preInitFields)
+        {
+            Debug.Assert(!type.IsCanonicalSubtype(CanonicalFormKind.Specific));
+            _type = type;
+
+            // sort the PreInitFieldInfo to appear in increasing offset order for easier emitting
+            _sortedPreInitFields = new List<PreInitFieldInfo>(preInitFields);
+            _sortedPreInitFields.Sort(FieldDescCompare);
+        }
+
+        static int FieldDescCompare(PreInitFieldInfo fieldInfo1, PreInitFieldInfo fieldInfo2)
+        {
+            return fieldInfo1.Field.Offset.AsInt - fieldInfo2.Field.Offset.AsInt;
+        }
+
+        protected override string GetName(NodeFactory factory) => this.GetMangledName(factory.NameMangler);
+
+        public void AppendMangledName(NameMangler nameMangler, Utf8StringBuilder sb)
+        {
+            sb.Append(GetMangledName(_type, nameMangler));
+        }
+
+        public int Offset => 0;
+        public MetadataType Type => _type;
+
+        public static string GetMangledName(TypeDesc type, NameMangler nameMangler)
+        {
+            return nameMangler.NodeMangler.GCStatics(type) + "__PreInitData";
+        }
+
+        public virtual bool IsExported(NodeFactory factory) => false; 
```

Comment:
If this always returns false, this shouldn't implement the `IExportableSymbolNode` interface.